src/twisted/application/twist/test/test_twist.py

```diff
@@ -201,3 +201,102 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class CrippledRunner(object):
+    def __init__(self, **kwargs):
+        self.args = kwargs
+
+    def run(self):
+        pass
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def startService(service):
+            pass
+
+        self.patch(MultiService, "startService", startService)
```

Comment:
This looks kind of sad.  What's going on?  I suggest refactoring something so you don't have to do this.